plugins/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/ui/CDEServer.java

```diff
@@ -0,0 +1,79 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.cdk.reddeer.ui;
+
+import org.jboss.reddeer.common.condition.AbstractWaitCondition;
+import org.jboss.reddeer.common.exception.WaitTimeoutExpiredException;
+import org.jboss.reddeer.common.logging.Logger;
+import org.jboss.reddeer.common.wait.TimePeriod;
+import org.jboss.reddeer.common.wait.WaitUntil;
+import org.jboss.reddeer.common.wait.WaitWhile;
+import org.jboss.reddeer.core.condition.JobIsRunning;
+import org.jboss.reddeer.core.condition.ShellWithTextIsAvailable;
+import org.jboss.reddeer.eclipse.wst.server.ui.view.Server;
+import org.jboss.reddeer.eclipse.wst.server.ui.view.ServersView;
+import org.jboss.reddeer.eclipse.wst.server.ui.view.ServersViewEnums.ServerState;
+import org.jboss.reddeer.swt.api.TreeItem;
+import org.jboss.reddeer.swt.impl.button.PushButton;
+import org.jboss.reddeer.swt.impl.menu.ContextMenu;
+import org.jboss.reddeer.swt.impl.shell.DefaultShell;
+
+public class CDEServer extends Server {
+	
+	private static Logger log = Logger.getLogger(CDEServer.class);
+	
+	private static final TimePeriod TIMEOUT = TimePeriod.VERY_LONG;
+	
+	public CDEServer(TreeItem item, ServersView view) {
+		super(item, view);
+	}
+	
+	@Override
+	protected void operateServerState(String menuItem, final ServerState resultState) {
+		log.debug("Operate server's state: '" + menuItem + "'");
+		select();
+		AbstractWaitCondition condition = new AbstractWaitCondition() {
+			
+			@Override
+			public boolean test() {
+				return resultState.equals(getLabel().getState());
+			}
+
+			@Override
+			public String description() {
+				return "server's state is: " + resultState.getText();
+			}
+		};
+		new ContextMenu(menuItem).select();
+		new WaitUntil(new JobIsRunning(), TIMEOUT);
+		if (resultState == ServerState.STARTING || resultState == ServerState.STARTED) {
+			confirmSSLCertificateDialog();
+		}
+		new WaitUntil(condition, TIMEOUT);
+		new WaitWhile(new JobIsRunning(), TIMEOUT);
+
+		//check if the server has expected state after jobs are done
+		//new WaitUntil(new ServerStateCondition(resultState), TIMEOUT);
+		new WaitUntil(condition, TIMEOUT);
```

Comment:
I believe it is. The very same sequence of wait conditions is used in parent's class method. I did not debug the code but it seems that sometimes intermediate job can change server's state: starting - started, stopping - stopped.